x-pack/plugins/infra/server/routes/services/lib/utils.ts

```diff
@@ -0,0 +1,22 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import { RouteValidationError, RouteValidationResultFactory } from '@kbn/core/server';
+
+type ValidateStringAssetFiltersReturn = [{ error: RouteValidationError }] | [null, any];
+
```

Comment:
this validation function makes sure the filters exist on the request and parses them, then we can continue validation of the filter object shape in the type validation